### ISSUE TITLE
Issue during Installation of SAM in Mojave

### ISSUE THREAD

**Issue by 06nenglish** [aws/homebrew-tap#16]
When trying to install the SAM CLI I get the following, as with other issues I've tried relinking Python and postinstalling python too which seemed to go OK.

The Full log of the install is below:
```
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
doitlive                          exploitdb                         handbrake                         libccd                            netcdf                            netpbm                            supervisor

==> Installing aws-sam-cli from aws/tap
==> Downloading https://files.pythonhosted.org/packages/2d/47/618e89211a10c3fc007949f84e88544722e8f9e43299ed28f188891e53fa/aws-sam-cli-0.10.0.tar.gz
Already downloaded: /Users/nathan/Library/Caches/Homebrew/downloads/c9038134ec8dbf4e1ebdb3491ab530a6198d809d15c5cbe600bcfa08b72e9a34--aws-sam-cli-0.10.0.tar.gz
==> Downloading https://files.pythonhosted.org/packages/4e/8b/75469c270ac544265f0020aa7c4ea925c5284b23e445cf3aa8b99f662690/virtualenv-16.1.0.tar.gz
Already downloaded: /Users/nathan/Library/Caches/Homebrew/downloads/bda8bcde19a275fd9c84a7ec957593976e90dfd0e461a87f5698f1d2286c0197--virtualenv-16.1.0.tar.gz
==> python3 -c import setuptools... --no-user-cfg install --prefix=/private/tmp/aws-sam-cli--homebrew-virtualenv-20181223-12622-8vgsat/target --install-scripts=/private/tmp/aws-sam-cli--homebrew-virtualenv-20181223-12622-8vgsat/target/bin
==> python3 -s /private/tmp/aws-sam-cli--homebrew-virtualenv-20181223-12622-8vgsat/target/bin/virtualenv -p python3 /usr/local/Cellar/aws-sam-cli/0.10.0/libexec
==> /usr/local/Cellar/aws-sam-cli/0.10.0/libexec/bin/pip install -v --no-binary :all: --ignore-installed /private/tmp/aws-sam-cli-20181223-12622-1qpbxpe/aws-sam-cli-0.10.0
Last 15 lines from /Users/nathan/Library/Logs/Homebrew/aws-sam-cli/03.pip:
Removed build tracker '/private/tmp/pip-req-tracker-7vq90frg'
Command "/usr/local/Cellar/aws-sam-cli/0.10.0/libexec/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/private/tmp/pip-install-80e99sw6/regex/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /private/tmp/pip-record-mlmbilcx/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/Cellar/aws-sam-cli/0.10.0/libexec/bin/../include/site/python3.7/regex" failed with error code 1 in /private/tmp/pip-install-80e99sw6/regex/
Exception information:
Traceback (most recent call last):
  File "/usr/local/Cellar/aws-sam-cli/0.10.0/libexec/lib/python3.7/site-packages/pip/_internal/cli/base_command.py", line 143, in main
    status = self.run(options, args)
  File "/usr/local/Cellar/aws-sam-cli/0.10.0/libexec/lib/python3.7/site-packages/pip/_internal/commands/install.py", line 366, in run
    use_user_site=options.use_user_site,
  File "/usr/local/Cellar/aws-sam-cli/0.10.0/libexec/lib/python3.7/site-packages/pip/_internal/req/__init__.py", line 49, in install_given_reqs
    **kwargs
  File "/usr/local/Cellar/aws-sam-cli/0.10.0/libexec/lib/python3.7/site-packages/pip/_internal/req/req_install.py", line 791, in install
    spinner=spinner,
  File "/usr/local/Cellar/aws-sam-cli/0.10.0/libexec/lib/python3.7/site-packages/pip/_internal/utils/misc.py", line 705, in call_subprocess
    % (command_desc, proc.returncode, cwd))
pip._internal.exceptions.InstallationError: Command "/usr/local/Cellar/aws-sam-cli/0.10.0/libexec/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/private/tmp/pip-install-80e99sw6/regex/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /private/tmp/pip-record-mlmbilcx/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/Cellar/aws-sam-cli/0.10.0/libexec/bin/../include/site/python3.7/regex" failed with error code 1 in /private/tmp/pip-install-80e99sw6/regex/

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/aws/homebrew-tap/issues
```

Any ideas or suggestions? My knowledge of Python and pip is next to nothing!

**Comment by mcnamee** [comment]
@06nenglish I had a similar issue - but found this issue which helped: https://github.com/awslabs/aws-sam-cli/issues/509

**Comment by 06nenglish** [comment]
Thanks @mcnamee That was actually very useful! As it narrowed down where the error was during the sam cli installation, which seems to be in `clang` (I haven't put the full stack trace for sanity)?

```
  In file included from regex_3/_regex.c:46:
  In file included from /usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/include/python3.7m/Python.h:34:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/stdlib.h:66:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/wait.h:110:
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:197:2: error: unknown type name 'uint64_t'
          uint64_t ri_user_time;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:198:2: error: unknown type name 'uint64_t'
          uint64_t ri_system_time;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:199:2: error: unknown type name 'uint64_t'
          uint64_t ri_pkg_idle_wkups;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:200:2: error: unknown type name 'uint64_t'
          uint64_t ri_interrupt_wkups;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:201:2: error: unknown type name 'uint64_t'
          uint64_t ri_pageins;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:202:2: error: unknown type name 'uint64_t'
          uint64_t ri_wired_size;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:203:2: error: unknown type name 'uint64_t'
          uint64_t ri_resident_size;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:204:2: error: unknown type name 'uint64_t'
          uint64_t ri_phys_footprint;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:205:2: error: unknown type name 'uint64_t'
          uint64_t ri_proc_start_abstime;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:206:2: error: unknown type name 'uint64_t'
          uint64_t ri_proc_exit_abstime;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:211:2: error: unknown type name 'uint64_t'
          uint64_t ri_user_time;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:212:2: error: unknown type name 'uint64_t'
          uint64_t ri_system_time;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:213:2: error: unknown type name 'uint64_t'
          uint64_t ri_pkg_idle_wkups;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:214:2: error: unknown type name 'uint64_t'
          uint64_t ri_interrupt_wkups;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:215:2: error: unknown type name 'uint64_t'
          uint64_t ri_pageins;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:216:2: error: unknown type name 'uint64_t'
          uint64_t ri_wired_size;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:217:2: error: unknown type name 'uint64_t'
          uint64_t ri_resident_size;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:218:2: error: unknown type name 'uint64_t'
          uint64_t ri_phys_footprint;
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/resource.h:219:2: error: unknown type name 'uint64_t'
          uint64_t ri_proc_start_abstime;
          ^
  fatal error: too many errors emitted, stopping now [-ferror-limit=]
  20 errors generated.
  error: command 'clang' failed with exit status 1
```
That then fails on the regex stuff that I had in my brew install error.

After a quick search for `unknown type name 'uint64_t'` I found [this](https://github.com/cython/cython/issues/2009), which pointed to an issue caused by migrating my old mac across to my new one. 

So after moving `/usr/local/include` to a safe location and force reinstalling python I was able to `brew install aws-sam-cli`!
